### ISSUE TITLE
[PFS mixin] Update interface for working with commits

### ISSUE THREAD

**Issue by albscui** [pachyderm/python-pachyderm#287]
# Background

Global IDs: https://github.com/pachyderm/rfcs/blob/global_ids/global_ids.md

The TLDR:
- A user can specify a commit in 3 ways
  - `repo, branch`
  - `repo, commit_id`
  - `repo, branch, commit_id`

# Problem:

Although Pachyderm uses gRPC + protobuf, the client should hide these details. Currently, our API leaks protobuf details by taking in protobuf objects and returning them.

# Solution
The solution is to implement adapters that are more Pythonic, but can be translated from and to protobuf objects.

# Design

Implement a `Commit` type that acts as the Pythonic version of `pfs_pb2.Commit`

```python
class Commit(NamedTuple):
    repo: str
    branch: str = None
    id: str = None
    repo_type: str = "user"

    def _to_pb(self) -> pfs_pb2.Commit:
    
    @staticmethod
    def _from_pb(commit: pfs_pb2.Commit) -> "Commit":

```

However, our APIs don't need a `Commit` object explicitly. In most cases, a `tuple` or `dict` can fully capture the information required for a `Commit`.

```python
# A CommitType can be a Commit, tuple, or dict
CommitType = Union[Commit, tuple, dict]

# Example interfaces
start_commit(repo, branch, parent: CommitType) -> Commit
flush_commit(commits: list[CommitType])
inspect_commit(commit: CommitType)
get_file(commit: CommitType, path: str)
```

**Comment by PFedak** [comment]
I don't know much about the python side of this, but worth noting that with system repos (see e.g. #6070), you might also need to specify a "type" field to access certain repos. My hope was that this is unnecessary for normal usage (the pach client defaults to the normal `user` type), but I can imagine wanting to access stats information from the python client.

**Comment by albscui** [comment]
> I don't know much about the python side of this, but worth noting that with system repos (see e.g. [pachyderm/pachyderm#6070](https://github.com/pachyderm/pachyderm/pull/6070) ), you might also need to specify a "type" field to access certain repos. My hope was that this is unnecessary for normal usage (the pach client defaults to the normal `user` type), but I can imagine wanting to access stats information from the python client. The CLI supports `repo.type@commit:path` now, too.

Ah I see, I'm defaulting to `user` type repos as well. I'm not sure if python-pachyderm users ever need to create any other types of repos. However, we can certainly support it!
